pkg/controller/operconfig/ingressconfig_controller.go

```diff
@@ -0,0 +1,90 @@
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// newIngressConfigReconciler returns a new reconcile.Reconciler
+func newIngressConfigReconciler(client client.Client) *ReconcileIngressConfigs {
+	return &ReconcileIngressConfigs{client: client}
+}
+
+var _ reconcile.Reconciler = &ReconcileIngressConfigs{}
+
+// ReconcileIngressConfigs watches for updates to ingress controller configuration
+// and sets the network policy related labels on the openshift-host-network namespace
+type ReconcileIngressConfigs struct {
+	client client.Client
+}
+
+// Reconcile sets the openshift-host-network namespaces' labels as per the
+// endpointPublishingStrategy of the `default` ingress controller object.
+func (r *ReconcileIngressConfigs) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	if request.Namespace == names.IngressControllerNamespace && request.Name == names.DefaultIngressControllerName {
+		log.Printf("Reconciling update to IngressController %s/%s\n", request.Namespace, request.Name)
+		ingressControllerConfig := &operv1.IngressController{TypeMeta: metav1.TypeMeta{APIVersion: operv1.GroupVersion.String(), Kind: "IngressController"}}
+		err := r.client.Get(context.TODO(), request.NamespacedName, ingressControllerConfig)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				log.Printf("Ingress Controller configuration %s was deleted", request.NamespacedName.String())
+				// Request object not found, could have been deleted after reconcile request.
+				// Owned objects are automatically garbage collected, since we set
+				// the ownerReference (see https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/).
+				// Return and don't requeue
+				return reconcile.Result{}, nil
+			}
+			// Error reading the object - requeue the request.
+			log.Printf("Unable to retrieve IngressController.operator.openshift.io object: %v", err)
+			return reconcile.Result{}, err
+		}
+		addLabel := ingressControllerConfig.Spec.EndpointPublishingStrategy != nil &&
+			ingressControllerConfig.Spec.EndpointPublishingStrategy.Type == operv1.HostNetworkStrategyType
+
+		err = r.updatePolicyGroupLabelOnNamespace(names.HostNetworkNamespace, addLabel)
+		if err != nil {
+			log.Printf("Error setting the host network label on namespace %s: %v", names.HostNetworkNamespace, err)
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{RequeueAfter: ResyncPeriod}, nil
+	}
+	return reconcile.Result{}, nil
+}
+
+// setLabelsOnNamespace sets the labels specified on the target namespace using the client API
+func (r *ReconcileIngressConfigs) updatePolicyGroupLabelOnNamespace(targetNamespace string, add bool) error {
+	var err error
+	namespace := &corev1.Namespace{TypeMeta: metav1.TypeMeta{APIVersion: corev1.SchemeGroupVersion.String(), Kind: "Namespace"}}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: targetNamespace}, namespace)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
```

Comment:
I continue to be unhappy about this; if we created the host-network namespace, and now it doesn't exist, then that's an error and should not just be ignored. (But if we _didn't_ create it, then we should not log any errors about it. Though, ideally we wouldn't even be here in that case.)

We at least need a FIXME